### ISSUE TITLE
Flutter unable to connect to VM service on iOS

### ISSUE THREAD

**Issue by yjbanov** [flutter/flutter#6077]
## Steps to Reproduce

Both `flutter run` and `flutter drive` are affected in both debug and `--profile` modes.

## Flutter Doctor

[✓] Flutter (on Mac OS, channel master)
    • Flutter at /Users/yjbanov/code/flutter/flutter
    • Framework revision d5b3588578 (27 minutes ago), 2016-09-26 13:00:54
    • Engine revision 3551e7a48e
    • Tools Dart version 1.20.0-dev.5.0

[✓] Android toolchain - develop for Android devices (Android SDK 23.0.3)
    • Android SDK at /Users/yjbanov/Library/Android/sdk
    • Platform android-23, build-tools 23.0.3
    • Java(TM) SE Runtime Environment (build 1.8.0_91-b14)

[✓] iOS toolchain - develop for iOS devices (Xcode 7.3.1)
    • XCode at /Applications/Xcode.app/Contents/Developer
    • Xcode 7.3.1, Build version 7D1014

[✓] Atom - a lightweight development environment for Flutter
    • flutter plugin version 0.2.5
    • dartlang plugin version 0.6.39

[✓] Connected devices
    • Emily Fortuna’s iPhone • f2943620b1e218437e4014a75d136d5121f199d9 • ios

## Logs and Crash Reports

```
$ flutter run
Launching loader on iPhone...
Exception from flutter run: SocketException: OS Error: Connection refused, errno = 61, address = 127.0.0.1, port = 62840
dart:io                                                           WebSocket.connect
package:flutter_tools/src/vmservice.dart 28                       VMService.connect.<async>
===== asynchronous gap ===========================
dart:async                                                        Future.Future.microtask
package:flutter_tools/src/vmservice.dart                          VMService.connect
package:flutter_tools/src/resident_runner.dart 106                ResidentRunner.connectToServiceProtocol.<async>
===== asynchronous gap ===========================
```

**Comment by yjbanov** [comment]
May be related to https://github.com/flutter/flutter/issues/6067 but log output is vastly different.

**Comment by cbracken** [comment]
This looks like a dupe of #6027, which was fixed in #6028.

**Comment by cbracken** [comment]
Fixed by #6090
